processor/k8sprocessor/pod_association.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8sprocessor
+
+import (
+	"context"
+	"net"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sprocessor/kube"
+
+	"go.opentelemetry.io/collector/client"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+)
+
+func k8sPodAssociationFromAttributes(ctx context.Context, attrs pdata.AttributeMap, associations kube.Associations) map[string]string {
+	var pod_ip, client_ip, context_ip string
+	podAssociation := make(map[string]string)
+
+	pod_ip = stringAttributeFromMap(attrs, k8sIPLabelName)
```

Comment:
I think this and the below should be associated within specific association name sections